### ISSUE TITLE
cargo and libressl support

### ISSUE THREAD

**Issue by semarie** [rust-lang/cargo#3428]
since `openssl-0.9.1`, cargo isn't buildable anymore with [libressl](http://www.libressl.org).

I dunno if it should be considered as a problem for cargo or openssl crate.

**Comment by steveklabnik** [comment]
What does "isn't buildable" mean? What errors are there? etc

I'd imagine that it's the openssl crate's bug, but hard to tell without knowing exactly what's broken.

**Comment by semarie** [comment]
sorry for the poor description (or the absence of). my bad.

yes, the origin is openssl crate: with 0.9 the crate starts to autodetect the OpenSSL version by parsing headers, and with LibreSSL (API compatible with OpenSSL 1.0.1, but not ABI compatible) it fails. I saw that it is a know issue https://github.com/sfackler/rust-openssl/issues/521.

Currently, I am try to workaround, but I dunno if the cargo binary will be fully functional or not (I just started a `make test`).

If it isn't, it could mean that cargo became unusable on platform using LibreSSL instead of OpenSSL.
